### PR TITLE
Issue44 doc why container

### PR DESCRIPTION
**Pull request by rqding**
Add the first draft of the doc of why use container

### CODE REVIEW

**Comment by lwjohnst86** [comment]
Trying to set up an Action to automatically reformat markdown text. Hasn't worked yet :stuck_out_tongue:

File: design/decisions/why-containers/Why_Container_Document.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Reason for Choosing Containerization Technology
+
+## What is Containerization
+
+Containerization is the technology of packaging software development code along with all of its necessary dependencies or configurations, including as libraries, frameworks, and other dependencies, into a single, isolated "container." The software or application inside the "container" can be transported, deployed and used reliably in any environment and on any infrastructure, regardless of the operating system. The container functions as a sort of computing environment or bubble around the software or application, keeping it apart from the actual computing environment. It made moving or deploying the software or application to any new computing environment less complicated.
+
+## Benefit of Containerization
```
Comment (K-Beicher) [inline design/decisions/why-containers/Why_Container_Document.md:7]: Do we also need a section where we look at the drawbacks compared with running a traditional server set-up.

Comment (rqding) [reply design/decisions/why-containers/Why_Container_Document.md:7]: Sure, I could add a section for "drawbacks of the Containerization"

**Comment by lwjohnst86** [comment]
Finally am back doing non-teaching work :laughing: 

Few things:

- Files inside the "why-" type folders, and in general, should be named "index.md".
- I've added some todo tags to add some other details, but for now I will merge this in.

**Review by lwjohnst86 — APPROVED** [review]

**Pull request merged**